darts/timeseries.py

```diff
@@ -904,6 +945,96 @@ def univariate_values(self, copy=True, sample=0) -> np.ndarray:
+            # of ``samples`` samples.
+        """
+
+        axis = TimeSeries._get_str_axis(axis)
+
+        display_n = range(min(size, self._xa.sizes[axis]))
+        return TimeSeries(self._xa[{axis: display_n}])
+
+    def tail(self,
+             size: Optional[int] = 5,
+             axis: Optional[Union[int, str]] = 'time') -> 'TimeSeries':
+        """
+        Return last n samples from TimeSeries.
+
+        Parameters
+        ----------
+        size : int, default: 5
+            number of samples to retain
+        axis : str or int, optional, default: 'time'
+            axis along which we intend to display records
+
+        Returns
+        -------
+        TimeSeries
+            Three dimensional array of (time, component, samples) where "samples" dimension has been cut off except
+            # of ``samples`` from the bottom. [Default: 5]
+        """
+
+        axis = TimeSeries._get_str_axis(axis)
+        display_n = range(-min(size, self._xa.sizes[axis]), 0)
+        return TimeSeries(self._xa[{axis: display_n}])
+
+    @staticmethod
+    def _get_str_axis(axis: Union[int, str]):
+        """Returns correct axis name as a string, regardless whether it was string or integer"""
+        axis_error_string = "Axis parameter can be only one of the numbers (0, 1, 2) " \
+                            "or strings ('time', 'component', 'sample')"
+        if isinstance(axis, int):
+            raise_if_not(0 <= axis <= 2, axis_error_string)
+            axis = DIMS[axis]
+        elif isinstance(axis, str):
+            raise_if(axis not in DIMS, axis_error_string)
+        else:
+            raise_log(AttributeError(axis_error_string))
+
+        return axis
+
+
+    def concatenate(self,
+                    other: 'TimeSeries',
+                    axis: Optional[Union[str, int]] = 0,
+                    ignore_time_axes: Optional[bool] = False):
+        """Concatenates another timeseries to the current one along given axis.
+
+            Note: when concatenating along the ``time`` dimension, first concatenated timeserie marks the start date of
+            the resulting series, and the remaining series will have their date indices overwritten.
+
+            Parameters
+            ----------
+            other : sequence of TimeSeries
+                another timeseries to concatenate to this one
+            axis : str or int
+                axis along which timeseries will be concatenated. ['time', 'component' or 'sample'; Default: 'time']
+            ignore_time_axes : bool, default False
+                Ignore errors when time axis varies for some timeseries. Note that this may yield unexpected results
+
+            Returns
+            -------
+            TimeSeries
+                concatenated timeseries
+        """
+        return concatenate(timeserie_sequence=[self, other], axis=axis, ignore_time_axes=ignore_time_axes)
```

Comment:
actually, since the new concatenate supports sequences of any length we could also concatenate a sequence of timeseries with `self` with something like this (and docs update):

```suggestion
        return concatenate(timeserie_sequence=[self] + other if isinstance(other, list) else [other], axis=axis, ignore_time_axes=ignore_time_axes)
```

Reply:
I think this is a bit overcomplicated.